pkg/archive/archive.go

```diff
@@ -722,6 +724,11 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, o
 	case tar.TypeFifo:
 		// Handle this is an OS-specific way
 		if err := handleTarTypeBlockCharFifo(hdr, path); err != nil {
+			if inUserns && errors.Is(err, syscall.EPERM) {
+				// In most cases, cannot create a fifo if running in user namespace
```

Comment:
Beside the previous code stumbled into this, is Fifo really something we don't expect to be working. In containerd we don't skip Fifo for user namespaces.

Reply:
I went a bit back-and-forth on both of these;

- changed all of `TypeFifo`,  `TypeBlock`, and `TypeChar` to try optimistically (but log the error on failure); this in the line of thought "who knows? one day it may work?"
- or the reverse; anticipating it won't work, so skip for `useNs`

In the end, I decided to keep the current behavior for now, but perhaps that's something we should revisit, WDYT?

Reply:
I think keep current behavior is right for the scope of this. Just the behavior difference was sticking out more so wanted to mention it.

Reply:
Yup, same! It also was a bit my motivation for moving the error-handling out of `handleTarTypeBlockCharFifo` - I wanted to make it more visible / explicit where we're ignoring errors.